### PR TITLE
Fix Gradient Propagation Through Dynamic Interventions

### PR DESCRIPTION
**Pull request by azane**
This was not working b/c timespans were being spliced with `torch.tensor` (which apparently clears out gradients) instead of `torch.cat` which preserves them. Two tests added — one of the behind-the-scenes torchdiffeq functionality, and another doing the exact same thing but using `DynamicIntervention`.

Note that gradients do not propagate through the `PointIntervention` `time` argument. I added a test for this but marked it as skipped for now. Added issue #228 to track this.

### CODE REVIEW

File: tests/dynamical/test_dynamic_interventions.py [hunk 1]
```diff
@@ -317,3 +323,151 @@ def test_split_twinworld_dynamic_matches_output(
...
+
+    # Z begins accruing dz=1 at t=param, so dzdparam should be -1.0.
+    (dzdparam,) = torch.autograd.grad(
+        outputs=(traj.z[-1],), inputs=(param,), create_graph=True
+    )
+    assert torch.isclose(dzdparam, tt(-1.0), atol=1e-5)
+
+
+@pytest.mark.skip(reason="TODO")
+def test_grad_of_point_intervention_params():
```
Comment (azane) [inline tests/dynamical/test_dynamic_interventions.py:368]: This is the skipped point intervention test. See issue #228.

**Review by agrawalraj — APPROVED** [review]
Great catch!
